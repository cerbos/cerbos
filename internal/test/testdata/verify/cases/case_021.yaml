description: Invalid YAML test suite with multiple principals and resources in expectations
