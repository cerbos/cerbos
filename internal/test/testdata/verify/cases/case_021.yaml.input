-- testdata/principals.yaml --
---
principals:
  bev: &bev
    id: bev
    policyVersion: '20210210'
    roles:
      - employee
      - manager
    attr:
      department: marketing
      geography: GB
      managed_geographies: GB
      ip_address: 10.20.1.2
      team: design
  barb:
    <<: *bev
    id: barb
  bob:
    <<: *bev
    id: bob

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr: &attr
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design
  pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr:
      <<: *attr
      status: PENDING_APPROVAL
  stale_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      <<: *attr
      modifiedAt: "2022-08-01T15:00:00Z"
  stale_pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      <<: *attr
      modifiedAt: "2022-08-01T15:00:00Z"
      status: PENDING_APPROVAL

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- suite_test.yaml --
---
name: TestSuite
description: Tests for verifying something
options:
  now: "2022-08-02T15:00:00Z"
tests:
  - name: No principal
    input:
      principals:
        - bev
        - barb
      resources:
        - pending_leave_request
      actions:
        - approve
        - delete
      auxData: myJWT
    expected:
      - resource: pending_leave_request
        actions:
          approve: EFFECT_ALLOW
          delete: EFFECT_ALLOW

  - name: Principal and principals
    input:
      principals:
        - bev
        - barb
      resources:
        - pending_leave_request
      actions:
        - approve
        - delete
      auxData: myJWT
    expected:
      - principal: bob
        principals:
          - bev
          - barb
        resource: pending_leave_request
        actions:
          approve: EFFECT_ALLOW
          delete: EFFECT_ALLOW

  - name: No resource
    input:
      principals:
        - bob
      resources:
        - pending_leave_request
        - stale_pending_leave_request
      actions:
        - approve
        - delete
      auxData: myJWT
    expected:
      - principal: bob
        actions:
          approve: EFFECT_ALLOW
          delete: EFFECT_ALLOW

  - name: Resource and resources
    input:
      principals:
        - bob
      resources:
        - pending_leave_request
        - stale_pending_leave_request
      actions:
        - approve
        - delete
      auxData: myJWT
    expected:
      - principal: bob
        resource: stale_leave_request
        resources:
          - pending_leave_request
          - stale_pending_leave_request
        actions:
          approve: EFFECT_ALLOW
          delete: EFFECT_ALLOW
