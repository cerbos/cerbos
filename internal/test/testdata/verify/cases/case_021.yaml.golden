{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "Unknown",
      "summary": {
        "overallResult": "RESULT_ERRORED"
      },
      "error": "failed to load test suite: validation error:\n - tests[0].expected[0]: either principal or principals must be present [expectation.principal_or_principals]\n - tests[1].expected[0]: principal and principals may not both be present [expectation.principal_nand_principals]\n - tests[2].expected[0]: either resource or resources must be present [expectation.resource_or_resources]\n - tests[3].expected[0]: resource and resources may not both be present [expectation.resource_nand_resources]"
    }
  ],
  "summary": {
    "overallResult": "RESULT_ERRORED"
  }
}
